### PR TITLE
Version number may be negative or greater than 255 #660

### PR DESCRIPTION
**Pull request by SteveSchafer-Innovent**
java.lang.IllegalArgumentException: Invalid version number: Version number may be negative or greater than 255 #660

Specify com.ibm.icu version 63.1 in target orbit R20190226160451 location and explicitly specify icu version every place it's included in ant build scripts (it doesn't appear in any maven dependencies).

Signed-off-by: Steve Schafer <sschafer@innoventsolutions.com>

### CODE REVIEW

**Comment by ruspl-afed** [comment]
LGTM

**Comment by wimjongman** [comment]
Why is it needed to explicitly name that version? Why would it not automatically use the highest version?

File: viewer/org.eclipse.birt.report.viewer/birt/WEB-INF/web.xml [hunk 1]
```diff
@@ -247,23 +247,23 @@ Automatically created by Apache Tomcat JspC.
 
 
     <servlet>
-        <servlet-name>org.apache.jsp.webcontent.birt.pages.common.Error_jsp</servlet-name>
-        <servlet-class>org.apache.jsp.webcontent.birt.pages.common.Error_jsp</servlet-class>
+        <servlet-name>org.apache.jsp.webcontent.birt.pages.common.processing_jsp</servlet-name>
```
Comment (wimjongman) [inline viewer/org.eclipse.birt.report.viewer/birt/WEB-INF/web.xml:250]: Steve, this does not seem related to issue #660, does it?

**Comment by ruspl-afed** [comment]
> Why is it needed to explicitly name that version? Why would it not automatically use the highest version?

I never heard about such kind of functionality for Apache ANT.

**Comment by wimjongman** [comment]
> > Why is it needed to explicitly name that version? Why would it not automatically use the highest version?
> 
> I never heard about such kind of functionality for Apache ANT.

I mean, why should this work

`				<include name="org.w3c.css.sac_*.jar" />`
`				<include name="com.ibm.icu_63.1.0.v20181030-1705.jar" />`

and not this:

`				<include name="org.w3c.css.sac_*.jar" />`
`				<include name="com.ibm.icu_*.jar" />`

**Comment by ruspl-afed** [comment]
> I mean, why should this work
> 
> ` <include name="org.w3c.css.sac_*.jar" />`
> ` <include name="com.ibm.icu_63.1.0.v20181030-1705.jar" />`
> 
> and not this:
> 
> ` <include name="org.w3c.css.sac_*.jar" />`
> ` <include name="com.ibm.icu_*.jar" />`

Yes, I understand. I have no idea which ICU4J jar will be used by ANT if we have several of them that matches the pattern. All of them are just files to be added to classpath, right?
Since we cannot guarantee that one and only one ICU4J jar will be available during classpath resolution - I think it is good idea to mention the full jar name explicitly.
Of cause this is not the best solution, since we "hardcoding" the exact version in the dozen of places - but this is the price we will pay for non-optimized build scripts that we currently have.

**Comment by ruspl-afed** [comment]
@wimjongman am I answered your question or should I try again?

**Comment by wimjongman** [comment]
> 
> 
> @wimjongman am I answered your question or should I try again?

Hi Alexander. I read your answer but I don't understand why there would be any other ICU version to copy than the most recent one.

**Comment by ruspl-afed** [comment]
> > @wimjongman am I answered your question or should I try again?
> 
> Hi Alexander. I read your answer but I don't understand why there would be any other ICU version to copy than the most recent one.

To be honest I don't know how many ICU jars we have available during the build. But I assume that after the "tycho part" of the build we _may_ have more than one ICU jar as a result of dependencies resolution.
Then we launch several ANT build that points to the directory with jars and says "pick an ICU jar for me, any version will match"
If we have several ICU jars there - which one will be used? To resolve this ambiguity the needed ICU jar specified precisely.

**Comment by wimjongman** [comment]
Steve is this still needed?

**Comment by wimjongman** [comment]
@SteveSchafer-Innovent is this still needed?

**Comment by wimjongman** [comment]
Closing because of no reaction fof @SteveSchafer-Innovent. Please reopen if needed.
